framework/transactions/handlers.py

```diff
@@ -52,7 +53,14 @@ def transaction_after_request(response, base_status_code_error=500):
     if view_has_annotation(NO_AUTO_TRANSACTION_ATTR):
         return response
     if response.status_code >= base_status_code_error:
-        commands.rollback()
+        original_exception = sys.exc_info()
+        try:
+            commands.rollback()
+        except OperationFailure:
+            logger.error('Transaction Rollback failed, original exception was: ', exc_info=original_exception)
```

Comment:
Can this be made to only display the error for the transaction rollback if it truly errors? Checking in the after_request/teardown_request depending on order of call to determine if was have not already done the rollback, even if we do decide to attempt it once again we in this case we don't need to report the error and can ignore it.